Cairo Desktop/Cairo Desktop/CairoSettingsWindow.xaml.cs

```diff
@@ -511,5 +519,283 @@ public static CairoSettingsWindow Instance
+        }
+
+        private void cboDesktopBackgroundType_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (cboDesktopBackgroundType.SelectedItem != null)
+            {
+                ComboBoxItem item = cboDesktopBackgroundType.SelectedItem as ComboBoxItem;
+                windowsImageBackgroundStackPanel.Visibility = (item.Tag == windowsImageBackgroundStackPanel) ? Visibility.Visible : Visibility.Collapsed;
+                cairoImageBackgroundStackPanel.Visibility = (item.Tag == cairoImageBackgroundStackPanel) ? Visibility.Visible : Visibility.Collapsed;
+                cairoVideoBackgroundStackPanel.Visibility = (item.Tag == cairoVideoBackgroundStackPanel) ? Visibility.Visible : Visibility.Collapsed;
+                bingImageBackgroundStackPanel.Visibility = (item.Tag == bingImageBackgroundStackPanel) ? Visibility.Visible : Visibility.Collapsed;
+
+                Settings.Instance.DesktopBackgroundType = item.Name;
+            }
+        }
+
+        #endregion
+
+        private void cboWindowsBackgroundStyle_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            // Stretched { WallpaperStyle = 2; TileWallpaper = 0 }
+            string wallpaperStyle = "2";
+            string tileWallpaper = "0";
+            switch (cboBingBackgroundStyle.SelectedItem)
+            {
+                case Desktop.CairoWallpaperStyle.Tile: // Tiled { WallpaperStyle = 0; TileWallpaper = 1 }
+                    wallpaperStyle = "0";
+                    tileWallpaper = "1";
+                    break;
+                case Desktop.CairoWallpaperStyle.Center: // Centered { WallpaperStyle = 1; TileWallpaper = 0 }
+                    wallpaperStyle = "1";
+                    tileWallpaper = "0";
+                    break;
+                case Desktop.CairoWallpaperStyle.Fit: // Fit { WallpaperStyle = 6; TileWallpaper = 0 }
+                    wallpaperStyle = "6";
+                    tileWallpaper = "0";
+                    break;
+                case Desktop.CairoWallpaperStyle.Fill: // Fill { WallpaperStyle = 10; TileWallpaper = 0 }
+                    wallpaperStyle = "10";
+                    tileWallpaper = "0";
+                    break;
+                case Desktop.CairoWallpaperStyle.Span: // Span { WallpaperStyle = 10; TileWallpaper = 0 }
+                    wallpaperStyle = "20";
+                    tileWallpaper = "0";
+                    break;
+            }
+
+            Registry.GetValue(@"HKEY_CURRENT_USER\Control Panel\Desktop", "WallpaperStyle", wallpaperStyle);
+            Registry.GetValue(@"HKEY_CURRENT_USER\Control Panel\Desktop", "TileWallpaper", tileWallpaper);
+
+
+            if (Shell.IsCairoUserShell)
+            {
+                // Tell DesktopWindow to ReloadBackground because we no settings are changing
+                Startup.DesktopWindow.ReloadBackground();
```

Comment:
Isn't this unnecessary since we already handle SPI.SETDESKWALLPAPER in the Desktop anyway?

Reply:
Looks like it should work that way

Reply:
Looks like both of the SPI_SETDESKWALLPAPERs are not setting the right style, only the path.

Reply:
I have only found documentation that shows setting style via Registry then calling SPI_SETDESKWALLPAPER. If we find documentation that states otherwise, we can change

Reply:
Hmm, for some reason I was seeing windows backgrounds style changes not working. I am an idiot and was selecting wallpaper that was the size of my screen. 🤦🏻‍♂️